data-table/table.js

```diff
@@ -1,46 +1,137 @@
-     */
+/**
+ * Functions for table creation
+ */
+
+/**
+ * If any option is not provided, chooses a sane default
+ * @param {object} clientConfig - A set of overriding config values, edits in-place
+ * @returns {undefined}         - Doesn't return anything
+ * @throws Error                - If any required option is not provided
+ */
+function setConfig(clientConfig) {
+
+    if (clientConfig.wrapperDivId === undefined) {
+        throw new Error("An ID for the wrapper div is required");
+    }
+
+    let config = new Config(clientConfig);
+
+    validateConfig(config);
+
+    configDict[clientConfig.wrapperDivId] = config;
 }
 
-function validateConfig(clientConfig) {
-    /**
-     * Performs any necessary logic checks on the configuration
-     * TODO: Fill this out
-     */
+/**
+ * Performs basic logic-checking on the config after defaults have been set
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
+ * @throws Error            - If any required option is not provided
+ */
+function validateConfig(config) {
+
+    if (config.numColumns <= 0 || config.numRows <= 0) {
+        throw new Error("The table must have at least one column and one row!")
+    }
 }
 
-function createEntryCell(indexString) {
-    /**
-     * Accesses the config to figure out what to make, then creates
-     * a cell at the specified spot in the HTML
-     * TODO: Fill this out
-     */
+/**
+ * Creates two sub divs and inserts them into the wrapper div
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
+ */
+function createSubDivs(config) {
+    let entryBoxDiv = document.createElement("div");
+    entryBoxDiv.id = config.tableIds.entryBoxDivId;
+
+    let tableDiv = document.createElement("div");
+    tableDiv.id = config.tableIds.tableDivId;
+
+    let text = document.createTextNode("~~This is where the entry box will live~~");
+    entryBoxDiv.appendChild(text);
+
+    document.getElementById(config.tableIds.wrapperDivId).appendChild(entryBoxDiv);
+    document.getElementById(config.tableIds.wrapperDivId).appendChild(tableDiv);
+}
+
+/**
+ * Creates an HTML table and assigns unique element IDs to all parts of the table.
+ * @param {object} config   - Table configuration object
+ * @returns {undefined}     - Doesn't return anything
+ */
+function createTable(config) {
+    let table = document.createElement("table");
+    table.id = config.tableIds.tableElementId;
+
+    let thead = document.createElement("thead");
+    thead.id = config.tableIds.theadElementId;
+
+    let tbody = document.createElement("tbody");
+    tbody.id = config.tableIds.tbodyElementId;
+
+    table.appendChild(tbody);
+    document.getElementById(config.tableIds.tableDivId).appendChild(table);
+    document.getElementById(config.tableIds.wrapperDivId).appendChild(document.getElementById(config.tableIds.tableDivId))
```

Comment:
[nit] Given that you call `document.getElementById(config.tableIds.tableDivId)` twice, you should be able to use:
```
let tableDivElement = document.getElementById(config.tableIds.tableDivId);
tableDivElement.appendChild(table);
document.getElementById(config.tableIds.wrapperDivId).appendChild(tableDivElement);
```

Reply:
Noted - will clean that up.